### PR TITLE
That Disagea 5 "for bad driver" shader fix

### PR DESCRIPTION
**Pull request by Gummygamer**

### CODE REVIEW

**Comment by ReaperOfSouls1909** [comment]
Testing now

**Comment by Gummygamer** [comment]
The story is that, even if I skipped the intro, Disagea 5 crashed, so, with the help of the main devs, I found a way to correct the shader to avoid that.

**Comment by ReaperOfSouls1909** [comment]
your fix is good it causes no issues and even fixed the one piece unlimited red driver crash as well great job

File: Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs [hunk 1]
```diff
@@ -297,7 +297,7 @@ private void PrintDeclAttributes(IEnumerable<ShaderDeclInfo> Decls, string InOut
             {
                 if (DeclInfo.Index >= 0)
                 {
-                    SB.AppendLine("layout (location = " + DeclInfo.Index + ") " + InOut + " " + GetDecl(DeclInfo) + ";");
+                    SB.AppendLine("layout (location = " + DeclInfo.Index + ") " + InOut + " " + "vec4 " + DeclInfo.Name + ";");
```
Comment (gdkchan) [inline Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs]: `+ " " + "vec4 " +` could be just `+ " vec4 " +`

Comment (Gummygamer) [reply Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs]: updated it.

File: Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs [hunk 2]
```diff
@@ -418,7 +418,7 @@ private void PrintAttrToOutput(string Identation = IdentationStr)
 
                 ShaderDeclInfo DeclInfo = KV.Value;
 
-                string Swizzle = ".xyzw".Substring(0, DeclInfo.Size + 1);
+                string Swizzle = ".xyzw";
```
Comment (ReinUsesLisp) [inline Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs]: If all attributes are `vec4` swizzle is not needed

Comment (Gummygamer) [reply Ryujinx.Graphics/Gal/Shader/GlslDecompiler.cs]: Eliminated those swizzles now.

**Comment by gdkchan** [comment]
I don't think that's what he meant. You don't need `.xyzw` if the type is aways a vec4, you can just assign directly.

**Comment by Gummygamer** [comment]
Erased those useless ".xyzw" too, sorry for the redundant code.

**Comment by ReinUsesLisp** [comment]
lgtm, but I didn't test it

**Comment by Gummygamer** [comment]
It seems to be fine in my test.

**Comment by ReinUsesLisp** [comment]
Just some small changes and it should be ready for me.
Remove `GetDecl` calls and replace them with `"vec4 " + DeclInfo.Name`. Also remove `ElemTypes` (because it would become unused).
I would have commented this in the code but GitHub doesn't let me comment unmodified code.

**Comment by Gummygamer** [comment]
A full elimination of ElemTypes and GetDecl causes an error because the shader still needs float for one case of declaration, I'm trying to put back the necessary evaluation in the best way possible.

**Pull request closed without merging**